prow/cmd/branchprotector/README.md

```diff
@@ -22,6 +22,8 @@ branch-protection:
           # Ensure that the extra-process-followed github status context passes.
           # In addition, adds any required prow jobs (aka always_run: true)
           require-contexts: ["extra-process-followed"]
+          # Tell Tide to merge when unspecified contexts are pending or failing
+          skip-unknown-contexts: true
```

Comment:
Configuration for `tide` actions should not live in the branch protector's config.

Reply:
+1, this should be part of the job spec

Reply:
For prow jobs we already have the optional setting. For optional jobs that are not prowjob we need to define it somewhere. I had a global setting in tide configuration, but that was not practical. It makes sense for me to define it in branch protection since this option only works when branch protection is set and you might want to apply this globally, on a repo or on a branch which branch protection already does. I had discussed this with @cjwagner

Reply:
also skipunknown context does not have any meaning if we don't have a branch protection, meaning no jobs defined at all in Prow config since all are unknown.

Reply:
So the original idea was to use the github required check to figure that out, but that was doing too many request to the github api. After a meeting with the fejta@, cjwagner@ and @BenTheElder it was decided to use branch protection instead. You don't have to actually enforce branch protection, you can just use the config and set protect to false, and then use that to define the required checks that you required for a PR to be merged. I don't see the point of duplicating those configs since they map back to protecting a branch. I think the Prow config is big and complicated enough.

Reply:
> You don't have to actually enforce branch protection, you can just use the config and set protect to false, and then use that to define the required checks that you required for a PR to be merged. 

Err, what? If you want to configure branch protection for statuses, do that. Then let `tide` honor required statuses. If you want to configure statuses for `tide`, do that, and don't touch the branch protection config (which you might have never even written since you don't use the branch protector). IDK. These seem like two entirely separate use cases that are being put in one place.

Reply:
I agree I am defining a tide options which will basically allow you to use branch protection or otherwise allow you to define your own if you don't want to use branch protection. If that option is not set, it will use combined status from github as it is today.